web/models/session.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Yggdrasil.Session do
+  alias Yggdrasil.User
+
+  def login(%{"username" => nil}, repo) do
+    :error
+  end
+
+  def login(params, repo) do
+    user = repo.get_by(User, username: String.downcase(params["username"]))
+    case authenticate(user, params["password"]) do
+      true -> {:ok, user}
+      _    -> :error
+    end
+  end
+
+  defp authenticate(_user, nil) do
+    false
+  end
+
+  defp authenticate(nil, _password) do
+    false
```

Comment:
I think you are suppose to use the [dummy_checkpw/0](http://hexdocs.pm/comeonin/Comeonin.Bcrypt.html#dummy_checkpw/0) in this case. It hashes `"password"` and always returns false to make it harder for username enumeration.